### PR TITLE
[Marksmanship] Sentinel module

### LINKED ISSUE #1069: [Marksmanship] Add Sentinel module
Some people use the talent because it has synergy with t214p, not that it's good - but might aswell support some sort of statistic on it.

### PR DESCRIPTION
**Pull request by Pewtro**
Adds a module for the Sentinel talent, and also showcases its buggy nature
image

Fixes #1069

### CODE REVIEW

**Comment by Pewtro** [comment]
Yeah it's a bug on blizzards end - wrote a larger post here: https://docs.google.com/document/d/1hi5Bdzpo6_ue6YH9mmmK_w5jIwP3fi3hM8dJXTX1zng/edit?usp=sharing

File: src/Parser/Hunter/Marksmanship/Modules/Talents/Sentinel.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  lastSentinelCastTimestamp = 0;
+  lostTicksFromCombatEnd = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.SENTINEL_TALENT.id);
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SENTINEL_TICK.id && spellId !== SPELLS.SENTINEL_TALENT.id) {
```
Comment (kfinch) [inline src/Parser/Hunter/Marksmanship/Modules/Talents/Sentinel.js]: Considering the `if` statements that follow, this `if` is entirely superfluous.

Comment (Pewtro) [reply src/Parser/Hunter/Marksmanship/Modules/Talents/Sentinel.js]: Makes sense - I'll just get rid of that one

**Comment by Pewtro** [comment]
You consume Hunter's Mark to cast Marked Shot on the target. 

Refreshing it on a target means you're losing marked shots on said target.

**Comment by Pewtro** [comment]
Yeah it consumes all in range (50+ yd) and shoots on them with that 1 global.

**Pull request merged**